### PR TITLE
Platform/eslint init setup

### PR DESCRIPTION
**Pull request by alienalien13**
This PR is mostly about eslint as a linter and its packages at all.
We should discuss rules.
#2

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+    "extends": [
+        "airbnb",
+        "react-app"
+    ],
+    "rules": {
+        "react/jsx-filename-extension": [1, {"extensions": [".js"]}],
+        "no-extra-semi": "error",
+        "semi": ["error", "never"]
+    }
+};
```
Comment (maybelinot) [inline .eslintrc.js]: new line? :D

**Comment by maybelinot** [comment]
Otherwise LGTM

**Comment by flashlee** [comment]
@alienalien13 I mentioned that you removed all semicolons and set semi rule as `never`. If @maybelinot has no objections to this (and I assume he doesn't, python 1 :heart:) , we should setup our eslint properly and attentively to avoid possible corner situations.
I suggest we can start here: https://eslint.org/docs/rules/semi 
Let's bring the following discussion [here](https://github.com/illWatchers/ill-watch-ui/issues/2)

**Comment by flashlee** [comment]
@alienalien13 btw what are the reasons behind your decision to drop semicolons?

**Comment by alienalien13** [comment]
@flashlee never have been using semicolons and I am dreaming to continue to do this :)

**Comment by alienalien13** [comment]
@maybelinot @flashlee 
Guys let's merge it and afterward we will discuss rules as an independent topic.
Now the rules including in this PR are just a sketch.

**Comment by flashlee** [comment]
@alienalien13 I want our taking decisions be healthy and meaningful. Sorry, but I can't see "never have been using semicolons" as meaningful reason. I've never written js code without semicolons. But I can't advocate for putting them back just because I don't have opposite experience. It wouldn't be reasonable.

By this day I can't find any opinion which explains in smart way why JS dev doesn't need semicolons.
Let's be honest and accept that fact that in reality JavaScript nowadays requires semicolons. And I'm talking not about ES specification, but about JavaScript compiler. It requires semicolons and that's why ASI put them into code before giving it to compiler. From this point I think our decision regarding semi should be based on question _How much can we rely on [ASI](http://inimino.org/~inimino/blog/javascript_semicolons)?_. The next question we should ask ourselves _How much can we relay on 3rd party implementation of ASI: transpilers (like babel) and minifiers (e.g. JSMin)?_. What minifier CRA has? Can we be sure that it will minify our code with proper semi insertion?(see Exmple 2 below)
There are examples of problems caused by omitting semicolons in real world. Let's see [Example 1](https://github.com/github/fetch/issues/420) and [Example 2](https://github.com/twbs/bootstrap/issues/3057). Second example is more interesting because JSMin didn't follow ASI implementation completely right (due to documentation `;` should be inserted before `!`).  And ESLint will not help in such situation because it is considering `[`, `(`, `/`, `+`, or `-`, but not `!`.
Designer of JS Brendan Eich wrote in his [blog post](https://brendaneich.com/2012/04/the-infernal-semicolon/)

>  ASI is (formally speaking) a syntactic error correction procedure.

And I see why. Because JS compiler sees omitting semicolons as a syntactic error. I know that if Brendan could he would change language spec and get rid of requirement of `;`. Maybe in future we will see something like this (most likely because of WebAssembly).
Because of everything above I can understand why, for example, [Google](https://google.github.io/styleguide/jsguide.html#formatting-semicolons-are-required) and [Airbnb](https://github.com/airbnb/javascript#semicolons) decided not to rely ASI and put semicolons. But I can't find any constructive point why it's good not to insert `;`. "Standart" eslint plugin README says:
>No semicolons – It's fine. Really!

Which I don't find very constructive.

Point of this commentary is to explain that in future I want our decisions be meaningful: properly investigated and discussed. I want that we all were familiar with "behind the scenes" of problem we are trying to solve and everyone of us know exactly why we are choosing this approach and not the other. Not to do something just because someone used to this or that.

File: .eslintrc.js (hunk 1 above)
Comment (flashlee) [reply .eslintrc.js]: I'm agree with @maybelinot. It's always a good practice to have a new line in the end of file

**Review by flashlee — APPROVED** [review]
Everything else looks good :smiley: 
LGTM :+1:

File: .eslintrc.js (hunk 1 above)
Comment (alienalien13) [reply .eslintrc.js]: Me either. However, it's automatically generated file, and we will adjust rules for sure, so we add it in next commit. And, we will set up a rule for this, as @flashlee mentioned here https://github.com/illWatchers/ill-watch-ui/issues/2#issuecomment-423937816

**Pull request merged**